staging/src/k8s.io/apiserver/pkg/server/options/audit.go

```diff
@@ -147,43 +147,43 @@ func (o *AuditOptions) Validate() []error {
 		}
 	} else {
 		// check webhook configuration
-		if err := validateBackendMode(pluginwebhook.PluginName, o.WebhookOptions.BatchOptions.Mode); err != nil {
-			allErrors = append(allErrors, err)
-		}
-		if err := validateBackendBatchConfig(pluginwebhook.PluginName, o.LogOptions.BatchOptions.BatchConfig); err != nil {
-			allErrors = append(allErrors, err)
+		if o.WebhookOptions.enabled() {
+			if err := validateBackendBatchOptions(pluginwebhook.PluginName, o.WebhookOptions.BatchOptions); err != nil {
+				allErrors = append(allErrors, err)
+			}
 		}
 
 		// check log configuration
-		if err := validateBackendMode(pluginlog.PluginName, o.LogOptions.BatchOptions.Mode); err != nil {
-			allErrors = append(allErrors, err)
-		}
-		if err := validateBackendBatchConfig(pluginlog.PluginName, o.LogOptions.BatchOptions.BatchConfig); err != nil {
-			allErrors = append(allErrors, err)
-		}
+		if o.LogOptions.enabled() {
+			if err := validateBackendBatchOptions(pluginlog.PluginName, o.LogOptions.BatchOptions); err != nil {
+				allErrors = append(allErrors, err)
+			}
 
-		// Check log format
-		validFormat := false
-		for _, f := range pluginlog.AllowedFormats {
-			if f == o.LogOptions.Format {
-				validFormat = true
-				break
+			// Check log format
+			validFormat := false
+			for _, f := range pluginlog.AllowedFormats {
+				if f == o.LogOptions.Format {
+					validFormat = true
+					break
+				}
+			}
+			if !validFormat {
+				allErrors = append(allErrors, fmt.Errorf("invalid audit log format %s, allowed formats are %q", o.LogOptions.Format, strings.Join(pluginlog.AllowedFormats, ",")))
 			}
-		}
-		if !validFormat {
-			allErrors = append(allErrors, fmt.Errorf("invalid audit log format %s, allowed formats are %q", o.LogOptions.Format, strings.Join(pluginlog.AllowedFormats, ",")))
 		}
 	}
 
-	// Check validities of MaxAge, MaxBackups and MaxSize of log options
-	if o.LogOptions.MaxAge < 0 {
-		allErrors = append(allErrors, fmt.Errorf("--audit-log-maxage %v can't be a negative number", o.LogOptions.MaxAge))
-	}
-	if o.LogOptions.MaxBackups < 0 {
-		allErrors = append(allErrors, fmt.Errorf("--audit-log-maxbackup %v can't be a negative number", o.LogOptions.MaxBackups))
-	}
-	if o.LogOptions.MaxSize < 0 {
-		allErrors = append(allErrors, fmt.Errorf("--audit-log-maxsize %v can't be a negative number", o.LogOptions.MaxSize))
+	// Check validities of MaxAge, MaxBackups and MaxSize of log options, if file log backend is enabled.
+	if o.LogOptions.enabled() {
```

Comment:
the whole block belongs into validateLogOptions

Reply:
Done.